testsuite/shared/src/main/java/org/jboss/as/test/integration/domain/management/util/DomainLifecycleUtil.java

```diff
@@ -575,6 +585,36 @@ public boolean isHostControllerStarted() {
         return false;
     }
 
+    /**
+     * @param hostname the hostname to query
+     * @param requiredState the {@link ControlledProcessState.State} expected, or null for any state}
+     * @return true if the host is present in the queried hosts's model (in the requiredState, if present), false otherwise.
+     */
+    public boolean isHostPresentInModel(final String hostname, final ControlledProcessState.State requiredState) throws Exception {
+        final ModelNode operation = new ModelNode();
+        operation.get(OP).set(READ_ATTRIBUTE_OPERATION);
+        operation.get(OP_ADDR).add(HOST, hostname);
+        operation.get(NAME).set(HOST_STATE);
+
+        ModelNode result;
+        try {
+            result = executeForResult(operation);
```

Comment:
Don't use executeForResult here. It just returns the value of {"result"=> ???} not the full response. As is this method will never return true. You're analyzing the full response here, so just do getDomainClient().execute(operation) and analyze that.

Hopefully with that the exception handling can be simpler; i.e. no catching Throwable, throwing Exception, converting Exception into InterruptedException.